src/PixiEditor/Models/DocumentModels/Public/DocumentOperationsModule.cs

```diff
@@ -234,7 +234,19 @@ public void ReplaceColor(Color oldColor, Color newColor)
     {
         if (Internals.ChangeController.IsChangeActive || oldColor == newColor)
             return;
+        
         Internals.ActionAccumulator.AddFinishedActions(new ReplaceColor_Action(oldColor, newColor));
+        ReplaceInPalette(oldColor, newColor);
+    }
+
+    private void ReplaceInPalette(Color oldColor, Color newColor)
+    {
+        int indexOfOldColor = Document.Palette.IndexOf(oldColor);
+        if(indexOfOldColor == -1)
+            return;
+        
+        Document.Palette.RemoveAt(indexOfOldColor);
+        Document.Palette.Insert(indexOfOldColor, newColor);
```

Comment:
A simple `Document.Palette[indexOfOldColor] = newColor;` should have done the job too

Reply:
oh, right lmao